### PR TITLE
Minor changes

### PR DESCRIPTION
**Pull request by Niklp09**
- Optimize textures w/ optipng
- Remove mt 0.4 compatibility stuff
- Other minor stuff (play sounds ephemeral, tweak style of some comments, ...)

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -65,25 +65,17 @@ local function is_player(player)
...
-	if player.get_meta then
-		local meta = player:get_meta()
-		if meta and value == nil then
-			meta:set_string(key, "")
-		elseif meta then
-			meta:set_string(key, tostring(value))
-		end
-	else
-		player:set_attribute(key, value)
+	local meta = player:get_meta()
```
Comment (SmallJoker) [inline init.lua:68]: Here and in the function below: `get_meta` is guaranteed to return a valid object for valid players. Checks against `meta` should not be necessary.

File: init.lua [hunk 2]
```diff
@@ -162,7 +154,7 @@ function stamina.set_poisoned(player, poisoned)
 		set_player_attribute(player, attribute.poisoned, "yes")
 	else
 		player:hud_change(hud_id, "text", "stamina_hud_fg.png")
-		set_player_attribute(player, attribute.poisoned, "no")
+		set_player_attribute(player, attribute.poisoned)
```
Comment (SmallJoker) [inline init.lua]: Passing an explicit `nil` (or `false`) would be more obvious in this case.

**Review by monk-afk — COMMENTED** [review]
The callback table iteration comes before the player:is_player() check.
A player receiving entity damage makes callback(player, change, cause) return with player (nil) and change (punch value)

https://github.com/minetest-mods/stamina/blob/9f1acb4eaaf6e5afbe8a343cd842e874d7b3fcf7/init.lua#L230

**Comment by Niklp09** [comment]
I know, but changing this would change the API behavior. Not a „minor change“ IMO

**Comment by SmallJoker** [comment]
Fine my me. Will merge in a few days unless there are objections.

File: init.lua [hunk 3]
```diff
@@ -125,7 +117,7 @@ function stamina.update_saturation(player, level)
 
 	local old = stamina.get_saturation(player)
 
-	if level == old then  -- To suppress HUD update
+	if level == old then -- To suppress HUD update
```
Comment (OgelGames) [inline init.lua:120]: FYI, this shouldn't have been changed, there should be two spaces before inline comments.

https://dev.minetest.net/Lua_code_style_guidelines#Comments

**Pull request merged**